pkg/executables/cmk.go

```diff
@@ -0,0 +1,363 @@
+	if err != nil {
+		return make([]types.CmkDiskOffering, 0), fmt.Errorf("error getting disk offerings info: %v", err)
+	}
+	if result.Len() == 0 {
+		return make([]types.CmkDiskOffering, 0), nil
+	}
+
+	response := struct {
+		CmkDiskOfferings []types.CmkDiskOffering `json:"diskoffering"`
+	}{}
+	err = json.Unmarshal(result.Bytes(), &response)
+	if err != nil {
+		return make([]types.CmkDiskOffering, 0), fmt.Errorf("failed to parse response into json: %v", err)
+	}
+	return response.CmkDiskOfferings, nil
+}
+
+// TODO: Add support for domain, account filtering
+func (c *Cmk) ListZones(ctx context.Context, offering string) ([]types.CmkZone, error) {
+	result, err := c.execWithNameAndIdFilters(ctx, offering, "list", "zones")
+	if err != nil {
+		return make([]types.CmkZone, 0), fmt.Errorf("error getting zones info: %v", err)
+	}
+	if result.Len() == 0 {
+		return make([]types.CmkZone, 0), nil
+	}
+
+	response := struct {
+		CmkZones []types.CmkZone `json:"zone"`
+	}{}
+	err = json.Unmarshal(result.Bytes(), &response)
+	if err != nil {
+		return make([]types.CmkZone, 0), fmt.Errorf("failed to parse response into json: %v", err)
+	}
+	return response.CmkZones, nil
+}
+
+// TODO: Add support for domain filtering
+func (c *Cmk) ListAccounts(ctx context.Context, account string) ([]types.CmkAccount, error) {
+	result, err := c.execWithNameAndIdFilters(ctx, account, "list", "accounts")
+	if err != nil {
+		return make([]types.CmkAccount, 0), fmt.Errorf("error getting accounts info: %v", err)
+	}
+	if result.Len() == 0 {
+		return make([]types.CmkAccount, 0), nil
+	}
+
+	response := struct {
+		CmkAccounts []types.CmkAccount `json:"account"`
+	}{}
+	err = json.Unmarshal(result.Bytes(), &response)
+	if err != nil {
+		return make([]types.CmkAccount, 0), fmt.Errorf("failed to parse response into json: %v", err)
+	}
+	return response.CmkAccounts, nil
+}
+
+func (c *Cmk) ListAffinityGroupsById(ctx context.Context, affinityGroupId string) ([]types.CmkAffinityGroup, error) {
+	idFilterParam := fmt.Sprintf("id=\"%s\"", affinityGroupId)
+	result, err := c.exec(ctx, "list", "affinitygroups", idFilterParam)
+	if err != nil {
+		return make([]types.CmkAffinityGroup, 0), fmt.Errorf("error getting affinity group info: %v", err)
+	}
+	if result.Len() == 0 {
+		return make([]types.CmkAffinityGroup, 0), nil
+	}
+
+	response := struct {
+		CmkAffinityGroups []types.CmkAffinityGroup `json:"affinitygroup"`
+	}{}
+	err = json.Unmarshal(result.Bytes(), &response)
+	if err != nil {
+		return make([]types.CmkAffinityGroup, 0), fmt.Errorf("failed to parse response into json: %v", err)
+	}
+	return response.CmkAffinityGroups, nil
+}
+
+func (c *Cmk) execWithNameAndIdFilters(ctx context.Context, parameterValue string, genericArgs ...string) (stdout bytes.Buffer, err error) {
+	argsWithNameFilterArg := append(genericArgs, fmt.Sprintf("name=\"%s\"", parameterValue))
+	argsWithIdFilterArg := append(genericArgs, fmt.Sprintf("id=\"%s\"", parameterValue))
```

Comment:
nit: can we build this only if we need it?

Reply:
you're suggesting I move this to line 255? I can do that, sure